### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by Ekristoffe**
Change: dependency from "snmpjs" to "net-snmp".
Add: SNMP version selection (v1, v2c, v3).
Add: v3 security.
New: IP filtering function with 2 different type of network mask.
New: log system (node-red use node log system, other use console log system with moment).
Bonus: it won't make node-red crash if the port can't be opened.
New: Help available inside the node information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,31 +3,92 @@ Node-RED SNMP Trap Listener
 
 [![NPM](https://nodei.co/npm/node-red-contrib-snmp-trap-listener.png)](https://nodei.co/npm/node-red-contrib-snmp-trap-listener/)
 
-This Node-RED node will bind itself to the configured port and start listening for SNMP Trap packages
-that meet the filter settings.
+This Node-RED node listen for SNMP Trap packages on the configured port.
```
Comment (ItsEcholot) [inline README.md]: `... node listens for ...`

File: README.md [hunk 2]
```diff
@@ -3,31 +3,92 @@ Node-RED SNMP Trap Listener
 
 [![NPM](https://nodei.co/npm/node-red-contrib-snmp-trap-listener.png)](https://nodei.co/npm/node-red-contrib-snmp-trap-listener/)
 
-This Node-RED node will bind itself to the configured port and start listening for SNMP Trap packages
-that meet the filter settings.
+This Node-RED node listen for SNMP Trap packages on the configured port.  
+Trap which do meet the filter setting will emitted on the node output.
```
Comment (ItsEcholot) [inline README.md]: `... will be emitted ...`

File: README.md [hunk 3]
```diff
@@ -3,31 +3,92 @@ Node-RED SNMP Trap Listener
...
 
-One can filter the received data by ip + netmask and/or by a community string.
+The filter option are:
```
Comment (ItsEcholot) [inline README.md]: `... filter options are:`

File: package.json [hunk 4]
```diff
@@ -1,21 +1,26 @@
...
+	],
+	"node-red": {
+		"nodes": {
+			"snmp-trap-listener": "snmp-trap-listener.js"
+		}
+	},
+	"author": [
+		"Marc Berchtold https://github.com/ItsEcholot/",
+		"Christophe Icard https://github.com/Ekristoffe/"
+	],
```
Comment (ItsEcholot) [inline package.json]: Based on the package.json doc (https://docs.npmjs.com/files/package.json#people-fields-author-contributors) there can only be one author. Please add yourself to the contributors array.

**Review by ItsEcholot — CHANGES_REQUESTED** [review]
Some small spelling changes I would like to request and the package.json author field is invalid right now. But apart from this very nice 👍 Thanks a lot for your work 🎉

File: README.md (hunk 1 above)
Comment (Ekristoffe) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (Ekristoffe) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (Ekristoffe) [reply README.md]: done

File: package.json (hunk 4 above)
Comment (Ekristoffe) [reply package.json]: done

**Comment by Ekristoffe** [comment]
I've done the requested changes.

**Review by ItsEcholot — APPROVED** [review]
Thanks for the changes

**Pull request merged**